nemo/collections/tts/losses/radttsloss.py

```diff
@@ -0,0 +1,215 @@
+# Copyright (c) 2021, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+###############################################################################
+import torch
+import torch.nn as nn
+from torch.nn import functional as F
+
+from nemo.collections.tts.helpers.common import get_mask_from_lengths
+
+
+def compute_flow_loss(z, log_det_W_list, log_s_list, n_elements, n_dims, mask, sigma=1.0):
```

Comment:
This is mostly defined elsewhere as well for waveglow, etc. Can you try to reuse that?

Reply:
The one in waveglow is somewhat different. it doesn't take n_elements and n_dims as an argument, doesn't also have loss_prior as an additional output. and other few differences.